config/runtime/server.go

```diff
@@ -406,6 +420,24 @@ func getBackendName(evalCtx *hcl.EvalContext, backendCtx hcl.Body) (string, erro
 	return "", nil
 }
 
+func splitBackendOrigin(evalCtx *hcl.EvalContext, backendCtx hcl.Body) (string, string, string) {
```

Comment:
In general splitting a url by scheme, host etc. could be achieved with `url.Parse(...)`. Lets move the origin setup to the health check initialization.